CRM/Utils/Check/Component/Schema.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | CiviCRM version 4.7                                                |
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC (c) 2004-2016                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC (c) 2004-2016
+ */
+class CRM_Utils_Check_Component_Schema extends CRM_Utils_Check_Component {
+
+  /**
+   * @return array
+   */
+  public function checkIndices() {
+    $messages = array();
+    $missingIndices = CRM_Core_BAO_SchemaHandler::getMissingIndices();
+    if ($missingIndices) {
```

Comment:
@totten @colemanw this is the almost-complete pr based on work by @aydun  - the only piece still missing is to turn this into something clickable "Please run System.updateindexes from the api explorer'"

I'm on the cusp of merging it without that piece as I'd like to get it merged but perhaps you have an easy suggestion - since it is an angular page.

Note that there will be some additional indexes in 4.7.19 that are declared in the schema. It is OK for this part to be in the following release as the theory is new installs get the new indexes on install, others on calling System.updateindexes api